### PR TITLE
[14.0][IMP] l10n_it_ricevute_bancarie: manage multiple bank in partner, migration script

### PR DESCRIPTION
**Pull request by GSLabIt**
Manage multiple bank

https://github.com/OCA/l10n-italy/issues/2707

### CODE REVIEW

**Review by OpenCode — APPROVED** [review]

**Comment by elvise** [comment]
@OCA/local-italy-maintainers merge ?

File: l10n_it_ricevute_bancarie/models/account.py [hunk 1]
```diff
@@ -71,9 +71,44 @@ def _compute_is_unsolved(self):
...
+            not self.riba_partner_bank_id
+            or self.riba_partner_bank_id not in self.partner_id.bank_ids
+        ):
+            bank_id = None
+            if (
+                self.partner_id
+                and self.invoice_payment_term_id.riba
+                and self.move_type in ["out_invoice", "out_refund"]
+            ):
+                bank_id = self.partner_id.mapped("bank_ids")
```
Comment (sergiocorato) [inline l10n_it_ricevute_bancarie/models/account.py:110]:
```suggestion
                bank_ids = self.partner_id.bank_ids
```

File: l10n_it_ricevute_bancarie/models/account.py [hunk 2]
```diff
@@ -71,9 +71,44 @@ def _compute_is_unsolved(self):
...
+            self.riba_partner_bank_id = bank_id[0] if bank_id else None
```
Comment (sergiocorato) [inline l10n_it_ricevute_bancarie/models/account.py:111]:
```suggestion
            self.riba_partner_bank_id = bank_ids[0] if bank_ids else None
```

File: l10n_it_ricevute_bancarie/models/account.py [hunk 3]
```diff
@@ -71,9 +71,44 @@ def _compute_is_unsolved(self):
...
+            invoice._onchange_riba_partner_bank_id()
+        return invoice
+
+    @api.onchange("partner_id", "invoice_payment_term_id", "move_type")
+    def _onchange_riba_partner_bank_id(self):
+        if (
+            not self.riba_partner_bank_id
+            or self.riba_partner_bank_id not in self.partner_id.bank_ids
+        ):
+            bank_id = None
```
Comment (sergiocorato) [inline l10n_it_ricevute_bancarie/models/account.py:104]:
```suggestion
            bank_ids = self.env['res.partner.bank']
```

**Review by sergiocorato — APPROVED** [review]
Some minor fixes

**Pull request merged**